### ISSUE TITLE
Weird progress indicators started appearing in stdout

### ISSUE THREAD

**Issue by martinstenhoff** [dotnet/msbuild#10998]
Why have I started seeing the following character sequence (which seems to some progress indicator) whenever I pipe dotnet program output to a file now?

```\x1b]9;4;3;\x1b\\\x1b]9;4;0;\x1b\\```

Program.cs:
```
Console.WriteLine("Hello, World!");
```

dotnet run > output.txt

output.txt:
```
]9;4;3;\]9;4;0;\Hello, World!
```

```
❯ dotnet --version
9.0.100-rc.2.24474.11
```
```
Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.100-rc.2.24474.11\
```

**Comment by MichalPavlik** [comment]
This will happen when using `dotnet run` with terminal that doesn't support ANSI escape codes. I prioritized this issue, and I will fix it soon.
Unfortunately, there is no way how to disable TL when starting `dotnet run`, right @baronfel?

**Comment by sailro** [comment]
@MichalPavlik:

- it is impacting terminals without ANSI escape code support indeed but also all redirected streams (like redirecting stdout to a file). Or applications using console to exchange data. It's really a breaking issue right now. 

- there is no workaround, both `--tl:off` or setting `MSBUILDTERMINALLOGGER ` to `off` are not working when using `dotnet run`

**Comment by MichalPavlik** [comment]
@sailro, that's right. I'm working on a fix that additionally respects the `MSBUILDTERMINALLOGGER` variable.

**Comment by baronfel** [comment]
@MichalPavlik some context:

There are up to three builds that happen during `dotnet run` currently:
* an implicit `msbuild /t:Restore` that is handled entirely by MSBuild
* an implicit `msbuild` (default Target) that is handled entirely by MSBuild
* an implicit `msbuild /t:Some_Target_Name_I_Forget` that is called by the CLI via the MSBuild API

The purpose of the third build is to be completely invisible _unless_ there are diagnostics reported from the third build. The third build's console logger setup should ideally be exactly matched to that of the first two builds, but the CLI doesn't _know_ what MSBuild's logging argument configuration is to attempt to recreate it via the API.  This is a long-running disjoint/mismatch between MSBuild and the CLI because MSBuild doesn't expose any knowledge about its argument parsing, but it's a particular pain point here for this use case.

So there are two requests here I think,:
* make sure that there's some programmatic way to ensure that the TL doesn't emit the 'processing' control codes
* see if there's a way to make it easier for the CLI to tell which console-based logger (Console or TL) we should use to stay in sync with the implicit restore and build that happens during `run`.

**Comment by willdean** [comment]
To me, additional confusion comes from this document: https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/sdk#terminal-logger

* It contains a list of commands that use the terminal logger by default which doesn't include `dotnet run`
* But then `dotnet run` docs show it _does_ take the --tl option and defaults to "auto", so the linked document should probably include `dotnet run`
* And then `dotnet run` doesn't (fully?) respect --tl anyway.

TBH I thought we'd given up spraying VT100 escape codes into output in the 1990s, I'm not sure why we have to go through this again.
